src/main/java/frc/utils/alerts/PeriodicAlert.java

```diff
@@ -0,0 +1,20 @@
+package frc.utils.alerts;
+
+import java.util.function.BooleanSupplier;
+
+public class PeriodicAlert extends Alert {
+
+	private final BooleanSupplier raiseSupplier;
+
+	public PeriodicAlert(AlertType type, String name, BooleanSupplier raiseSupplier) {
+		super(type, name);
+		this.raiseSupplier = raiseSupplier;
+	}
+
+	protected void reportByCondition() {
```

Comment:
consider overriding the reportAlert function instead

Reply:
i prefer to not override it, no reason to cancel the option to just report normally

Reply:
Why would you report it when the supplier isn't true? That seems wrong

Reply:
overriding makes sense because it is the essence of periodic alert, if wanted to report manually there is regular alert

Reply:
but why remove the normal option

Reply:
Because you shouldn't report something that is not happening